packages/react-components/src/components/timegraph-output-component.tsx

```diff
@@ -216,20 +278,49 @@ export class TimegraphOutputComponent extends AbstractTreeOutputComponent<Timegr
     private doHandleMarkerFilterChangedSignal(annotationMarkers: string[]) {
         this.annotationMarkers = annotationMarkers;
         this.chartLayer.updateChart();
+        this.markersChartLayer.updateChart();
+    }
+
+    private getMarkersLayerHeight() {
+        return (this.state.annotationEntries.length <= 1 ? 0 : (this.state.collpasedAnnotationNodes.length ? 1 :
+            this.state.annotationEntries.length) + 0.1) * 20;
 
     }
 
     renderTree(): React.ReactNode {
         // TODO Show header, when we can have entries in-line with timeline-chart
-        return <EntryTree
-            collapsedNodes={this.state.collapsedNodes}
-            showFilter={false}
-            entries={this.state.timegraphTree}
-            showCheckboxes={false}
-            onToggleCollapse={this.onToggleCollapse}
-            showHeader={false}
-            className="timegraph-tree"
-        />;
+        return <>
+            <div ref={this.timeGraphTreeRef} onScroll={_ev => { this.synchronizeTreeScroll(); }}
+                className='output-component-tree'
+                style={{
+                    height: parseInt(this.props.style.height.toString()) - this.getMarkersLayerHeight()
+                }}>
+                <EntryTree
+                    collapsedNodes={this.state.collapsedNodes}
+                    showFilter={false}
+                    entries={this.state.timegraphTree}
+                    showCheckboxes={false}
+                    onToggleCollapse={this.onToggleCollapse}
+                    showHeader={false}
+                    className="timegraph-tree"
+                />
+            </div>
+            <div ref={this.markerTreeRef} onScroll={_ev => { this.synchronizeTreeScroll(); }}
+                className='output-component-tree'
+                style={{ height: this.getMarkersLayerHeight() }}>
+                <EntryTree
+                    collapsedNodes={this.state.collpasedAnnotationNodes}
+                    showFilter={false}
+                    entries={this.state.annotationEntries}
```

Comment:
I'm not sure where it happens, but it seems to me like the close button and category label are centered horizontally in the tree. I think they should be left- or right- aligned.